### PR TITLE
feat: Add buf-language-server support

### PR DESCRIPTION
**Pull request by rinx**
Signed-off-by: Rintaro Okamura <rintaro.okamura@gmail.com>

This PR adds support for [buf-language-server](https://github.com/bufbuild/buf-language-server).

### CODE REVIEW

**Comment by justinmk** [comment]
failing CI

```
skip=./doc/server_configurations.md,./doc/server_configurations.txt --ignore-words=.codespellignorewords
./lua/lspconfig/server_configurations/nickel_ls.lua:33: prequisite ==> prerequisite
./test/lspconfig_spec.lua:3[5](https://github.com/neovim/nvim-lspconfig/runs/7917557364?check_suite_focus=true#step:4:6): exsts ==> exists
./test/lspconfig_spec.lua:79: exsts ==> exists

**Comment by rinx** [comment]
@justinmk I think the failure is not caused by the changes in this Pull Request.

The message indicates `./lua/lspconfig/server_configurations/nickel_ls.lua` and `./test/lspconfig_spec.lua` have misspelling words, but this PR doesn't include changes for these files.
https://github.com/neovim/nvim-lspconfig/runs/7917557364?check_suite_focus=true

**Comment by rinx** [comment]
Recently, codespell is updated to [v2.2.0](https://github.com/codespell-project/codespell/releases/tag/v2.2.0) and the release includes these misspelled words `prequisite` and `exsts`.

**Comment by rinx** [comment]
Thanks!

File: lua/lspconfig/server_configurations/bufls.lua [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+if vim.fn.has 'win32' == 1 then
+  cmd = { 'cmd.exe', '/C', bin_name, 'start' }
+end
+
+return {
+  default_config = {
+    cmd = cmd,
+    filetypes = { 'proto' },
+    root_dir = function(fname)
+      return util.root_pattern('buf.work.yaml', '.git')(fname)
```
Comment (stefanvanburen) [inline lua/lspconfig/server_configurations/bufls.lua:15]: @rinx should we also include 'buf.yaml' here, for buf modules that don't use workspaces?

Comment (rinx) [reply lua/lspconfig/server_configurations/bufls.lua:15]: @stefanvanburen Thanks for pointing it out. You're right!
I forgot to add it. 😓

**Pull request merged**